openedx/core/djangoapps/user_api/preferences/api.py

```diff
@@ -0,0 +1,179 @@
+
+
+class PreferenceRequestError(Exception):
+    """There was a problem with a preference request."""
+    pass
+
+
+class PreferenceNotFound(PreferenceRequestError):
+    """The desired user preference was not found."""
+    pass
+
+
+def _get_user(requesting_user, username, allow_staff=False):
+    """
+    Helper method to return the user for a given username.
+    """
+    try:
+        existing_user = User.objects.get(username=username)
+    except ObjectDoesNotExist:
+        raise AccountUserNotFound()
+
+    if requesting_user.username != username:
+        if not requesting_user.is_staff or not allow_staff:
+            raise AccountNotAuthorized()
+
+    return existing_user
+
+
+def get_user_preference(requesting_user, preference_key, username=None):
+    """Returns the value of the user preference with the specified key.
+
+    Args:
+        requesting_user (User): The user requesting the user preferences. Only the user with username
+            `username` or users with "is_staff" privileges can access the preferences.
+        preference_key (string): The key for the user preference.
+        username (str): Optional username for which to look up the preferences. If not specified,
+            `requesting_user.username` is assumed.
+
+    Returns:
+         The value for the user preference.
+
+    Raises:
+         AccountUserNotFound: no user with username `username` exists (or `requesting_user.username` if
+            `username` is not specified)
+        AccountNotAuthorized: the requesting_user does not have access to the user preference.
+        PreferenceNotFound: the user does not have a preference with the specified key.
+    """
+    existing_user = _get_user(requesting_user, username, allow_staff=True)
+    try:
+        preference_value = existing_user.preferences.get(key=preference_key).value
+    except ObjectDoesNotExist:
+        raise PreferenceNotFound()
+    return preference_value
+
+
+def get_user_preferences(requesting_user, username=None):
+    """Returns all user preferences as a JSON response.
+
+    Args:
+        requesting_user (User): The user requesting the user preferences. Only the user with username
+            `username` or users with "is_staff" privileges can access the preferences.
+        username (str): Optional username for which to look up the preferences. If not specified,
+            `requesting_user.username` is assumed.
+
+    Returns:
+         A dict containing account fields.
+
+    Raises:
+         AccountUserNotFound: no user with username `username` exists (or `requesting_user.username` if
+            `username` is not specified)
+        AccountNotAuthorized: the requesting_user does not have access to the user preference.
+    """
+    existing_user = _get_user(requesting_user, username, allow_staff=True)
+    user_serializer = UserSerializer(existing_user)
+    return user_serializer.data["preferences"]
```

Comment:
Use UserSerializer. get_preferences? I'm not really sure what the difference is (if there is one...).